dotnet/src/SemanticKernel.Abstractions/AI/ChatCompletion/ChatHistory.cs

```diff
@@ -14,53 +18,85 @@ public enum AuthorRoles
         Assistant = 2,
     }
 
+    private sealed class ChatMessage : ChatMessageBase
+    {
+        public ChatMessage(AuthorRole authorRole, string content) : base(authorRole, content)
+        {
+        }
+    }
+
     /// <summary>
     /// Chat message representation
     /// </summary>
-    public class Message
+    [Obsolete("This class is deprecated, using instances of this class will not be supported")]
+    public class Message : ChatMessageBase
     {
         /// <summary>
         /// Role of the message author, e.g. user/assistant/system
         /// </summary>
         public AuthorRoles AuthorRole { get; set; }
 
-        /// <summary>
-        /// Message content
-        /// </summary>
-        public string Content { get; set; }
-
         /// <summary>
         /// Create a new instance
         /// </summary>
         /// <param name="authorRole">Role of message author</param>
         /// <param name="content">Message content</param>
-        public Message(AuthorRoles authorRole, string content)
+        public Message(AuthorRoles authorRole, string content) : base(new AuthorRole(authorRole.ToString()), content)
```

Comment:
`AuthorRoles` enum has 4 values `Unknown`, `System`, `User`, `Assistant`, starting with uppercase letter. `.ToString()` method of enum value will return the same string with first uppercase letter. 

While your new struct `AuthorRole` has 3 pre-configured roles with labels starting with lowercase letter: `system`, `assistant`, `user`. It looks like new approach is not consistent with old approach. 

Maybe from OpenAI API perspective it's not a problem, just want to make sure it won't introduce any bugs related to case sensitivity.

Reply:
Good point, I will move those to be capitalized as the old approach.